shared/js/background/events.es6.js

```diff
@@ -266,35 +266,46 @@ startup.ready().then(() => {
     )
 })
 
+// Store the created tab id for when onBeforeNavigate is called so data can be copied across from the source tab
+const createdTargets = new Map()
 browser.webNavigation.onCreatedNavigationTarget.addListener(details => {
-    const currentTab = tabManager.get({ tabId: details.sourceTabId })
-    if (currentTab && currentTab.adClick) {
-        const newTab = tabManager.createOrUpdateTab(details.tabId, { url: details.url })
-        if (currentTab.adClick.shouldPropagateAdClickForNewTab(newTab)) {
-            newTab.adClick = currentTab.adClick.propagate(newTab.id)
-        }
-    }
+    createdTargets.set(details.tabId, details.sourceTabId)
```

Comment:
I assume this will be short-lived enough that in MV3 the service-worker lifecycle won't break this?

Reply:
Yeah it's literally between calls the risk is minimal enough I think we can accept it. Happy to switch to the session storage helper if you're worried?